### PR TITLE
Ssl docs

### PR DESCRIPTION
**Pull request by moshe010**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -67,18 +67,79 @@ $ DOCKERFILE=myfile TAG=mytag make image
...
+$ docker restart ufm
+```
+
+Create UFM secret
+```
+$ kubectl create secret generic ib-kubernetes-ufm-secret --namespace="kube-system" --from-literal=UFM_USER="admin" --from-literal=UFM_PASSWORD="12345" --from-literal=UFM_ADDRESS="127.0.01" --from-file=UFM_CERTIFICATE=ufmlocalhost.crt --dry-run -o yaml > ib-kubernetes-ufm-secret.yaml
+$ kubectl create -f ./ib-kubernetes-ufm-secret.yaml 
+```
+
+UFM on bare-metal
```
Comment (adrianchiris) [inline README.md]: why does ufm on container restart before secret creation and ufm on baremetal after ?

can we have both restarts then the secret creation ?

i.e
UFM restart
  containerized: ...
  baremetal: ....
Create secret ....

File: README.md [hunk 2]
```diff
@@ -67,18 +67,79 @@ $ DOCKERFILE=myfile TAG=mytag make image
...
+  name: ib-kubernetes-ufm-secret
+  namespace: kube-system
+stringData:
+  UFM_USERNAME: "admin"  # UFM Username
+  UFM_PASSWORD: "123456" # UFM Password
+  UFM_ADDRESS: ""        # UFM Hostname/IP Address 
+  UFM_HTTP_SCHEMA: ""    # http/https. Default: httpd
+  UFM_PORT: ""           # UFM REST API port: https default 443 and http default 80
+string:
+  UFM_CERTIFICATE: ""    # UFM Certificate in bas64 format . (if not provided client crtificate verfication is skipt)
```
[the remark this answers is not in the document]
Comment (moshe010) [reply README.md]: sorry we are verifying the server certificate

File: README.md [hunk 3]
```diff
@@ -41,49 +41,123 @@ Upon successful build the binary will be available in `build/ib-kubernetes`.
...
+#### Self Signed Certificates
 
-**Configurations if "ufm" subnet manager plugin is used for  `deployment/ib-kubernetes-ufm-secret.yaml`:**
-* `UFM_USERNAME`: Username of UFM.
-* `UFM_PASSWORD`: Password of UFM.
-* `UFM_ADDRESS`: IP address or hostname of UFM server.
-* `UFM_HTTP_SCHEMA`: http/https, default is https.
-* `UFM_PORT`: REST API port of UFM default is 443 (https), if `httpSchema` is set to http then the default is 80.
-* `UFM_CERTIFICATE`: Secure certificate if using secure connection.
+Login to UFM
```
Comment (adrianchiris) [inline README.md]: it would be nice if we headline the stages e.g 

##### Login to UFM

File: README.md [hunk 4]
```diff
@@ -41,49 +41,123 @@ Upon successful build the binary will be available in `build/ib-kubernetes`.
...
+
+containerized UFM:
+``` 
+$ docker exec -it ufm bash
+```
+
+Create private key and certificate
```
Comment (adrianchiris) [inline README.md]: ##### Create private key and certificate

File: README.md [hunk 5]
```diff
@@ -41,49 +41,123 @@ Upon successful build the binary will be available in `build/ib-kubernetes`.
...
+```
+$ openssl req -x509 -newkey rsa:4096 -keyout ufm.key -out ufm.crt -days 365 -subj '/CN=<UFM hostname>'
+```
+
+Install UFM private key and certificate
```
Comment (adrianchiris) [inline README.md]: ##### Install UFM private key and certificate

File: README.md [hunk 6]
```diff
@@ -41,49 +41,123 @@ Upon successful build the binary will be available in `build/ib-kubernetes`.
...
+```
+$ cp ufm.key /etc/pki/tls/private/ufmlocalhost.key
+$ cp ufm.crt /etc/pki/tls/certs/ufmlocalhost.crt
+```
+
+Restart UFM
```
Comment (adrianchiris) [inline README.md]: ##### Restart UFM

File: README.md [hunk 7]
```diff
@@ -41,49 +41,123 @@ Upon successful build the binary will be available in `build/ib-kubernetes`.
...
+```
+$ docker restart ufm
+```
+
+bare-metal UFM:
+```
+systemctl restart ufmd
+```
+
+Create UFM Plugin secret
```
Comment (adrianchiris) [inline README.md]: ##### Create UFM Plugin secret

File: README.md [hunk 8]
```diff
@@ -10,6 +10,9 @@
          * [Building Subnet Manager Plugins](#building-subnet-manager-plugins)
          * [Building Container Image](#building-container-image)
       * [Configuration Reference](#configuration-reference)
+      * [Plugins](#plugins)
+         * [NOOP Plugin](#noop-plugin)
```
Comment (adrianchiris) [inline README.md]: nit: below its Noop and here its NOOP, should be consistent

File: README.md [hunk 9]
```diff
@@ -41,54 +44,132 @@ Upon successful build the binary will be available in `build/ib-kubernetes`.
...
 
-# building one plugin, make <plugin-name>-plugin
-$ make noop-plugin
+#### Building a specific plugin
+```
+make <plugin name>-plugin
+```
+Example:
+```
+$ make <>-plugin
```
Comment (adrianchiris) [inline README.md]:
```
$ make <plugin-name>-plugin
```

File: README.md [hunk 10]
```diff
@@ -41,54 +44,132 @@ Upon successful build the binary will be available in `build/ib-kubernetes`.
...
+  UFM_USERNAME: "admin"  # UFM Username
+  UFM_PASSWORD: "123456" # UFM Password
+  UFM_ADDRESS: ""        # UFM Hostname/IP Address 
+  UFM_HTTP_SCHEMA: ""    # http/https. Default: https
+  UFM_PORT: ""           # UFM REST API port. Defaults: 443(https), 80(http)
+string:
+  UFM_CERTIFICATE: ""    # UFM Certificate in base64 format. (if not provided client will not verify server's certificate chain and host name)
+```
+
+#### Self Signed Certificates
```
Comment (adrianchiris) [inline README.md]: Suggestion, perhaps add a preface explaining why is this needed.

e.g
UFM utilizes certificates to authenticate requests, during deployment you should provide UFM with a valid certificate in your organization or create a self signed one.

what do you think ?

File: README.md [hunk 11]
```diff
@@ -41,54 +44,132 @@ Upon successful build the binary will be available in `build/ib-kubernetes`.
...
+``` 
+$ docker exec -it ufm bash
+```
+
+##### Create private key and certificate
+```
+$ openssl req -x509 -newkey rsa:4096 -keyout ufm.key -out ufm.crt -days 365 -subj '/CN=<UFM hostname>'
+```
+
+##### Install UFM private key and certificate
```
Comment (adrianchiris) [inline README.md]: the installation of certificate is regardless to when self-signed are used or provided by the user right ?
so perhaps move the headline back i.e 4x#

Comment (moshe010) [reply README.md]: I restructure it, let me know what you think

File: README.md [hunk 12]
```diff
@@ -41,54 +44,148 @@ Upon successful build the binary will be available in `build/ib-kubernetes`.
...
+#### Install UFM private key and certificate
+
+##### Login to UFM
+
+Containerized UFM:
+``` 
+$ docker exec -it ufm bash
+```
+
+##### cp private and crtificate to UFM location
```
Comment (adrianchiris) [inline README.md:155]: as its a healine  id suggest to replace cp with Copy

File: README.md [hunk 13]
```diff
@@ -41,54 +44,148 @@ Upon successful build the binary will be available in `build/ib-kubernetes`.
...
+```
+$ cp ufm.key /etc/pki/tls/private/ufmlocalhost.key
+$ cp ufm.crt /etc/pki/tls/certs/ufmlocalhost.crt
+
+```
+
+######  Restart UFM
```
Comment (adrianchiris) [inline README.md:162]: should be 5 X #  as a direct child under Install UFM private key and certificate ?

File: README.md [hunk 14]
```diff
@@ -41,54 +44,148 @@ Upon successful build the binary will be available in `build/ib-kubernetes`.
...
+```
+$ docker restart ufm
+```
+
+Bare-metal UFM:
+```
+systemctl restart ufmd
+```
+
+##### Create UFM secret
```
Comment (adrianchiris) [inline README.md:174]: a suggestion, this can be a direct child under UFM ( # X 4)
because you need to create secret in all cases.
